cpp/src/dataframe/device_table.cuh

```diff
@@ -227,6 +228,96 @@ __device__ inline bool rows_equal(device_table const& lhs,
                        equal_elements);
 }
 
+namespace {
+enum class State {False = 0, True = 1, Undecided = 2};
+
+struct typed_inequality_comparator {
+  template<typename ColType>
+	  __device__
```

Comment:
Indentation is all mixed up in this file. Can you please stick to spaces?

Reply:
fixed